src/Foundatio/Serializer/MessagePackSerializer.cs

```diff
@@ -6,17 +6,29 @@
 namespace Foundatio.Serializer {
     public class MessagePackSerializer : ISerializer {
         private readonly IFormatterResolver _formatterResolver;
+        private readonly bool _compressEnabled;
 
-        public MessagePackSerializer(IFormatterResolver resolver = null) {
+        public MessagePackSerializer(IFormatterResolver resolver = null, bool compressEnabled = false) {
+            _compressEnabled = compressEnabled;
             _formatterResolver = resolver ?? ContractlessStandardResolver.Instance;
         }
 
         public void Serialize(object data, Stream output) {
-            MessagePack.MessagePackSerializer.NonGeneric.Serialize(data.GetType(), output, data, _formatterResolver);
+            if (_compressEnabled) {
+                MessagePack.LZ4MessagePackSerializer.NonGeneric.Serialize(data.GetType(), output, data, _formatterResolver);
+            }
+            else {
+                MessagePack.MessagePackSerializer.NonGeneric.Serialize(data.GetType(), output, data, _formatterResolver);
+            }
         }
 
         public object Deserialize(Stream input, Type objectType) {
-            return MessagePack.MessagePackSerializer.NonGeneric.Deserialize(objectType, input, _formatterResolver);
+            if (_compressEnabled) {
```

Comment:
On a side note, I need to work on coding style guide but we try to have single statements not have braces and K&R coding style (everything on same line `} else {`